core/commands/swarm.go

```diff
@@ -812,3 +814,72 @@ func filtersRemove(r repo.Repo, cfg *config.Config, toRemoveFilters []string) ([
 
 	return removed, nil
 }
+
+var swarmConnMgrCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Interact with the Connection Manager.",
+		ShortDescription: `
+Display information about the current state of the connection manager.
+`,
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		n, err := req.InvocContext().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		if n.PeerHost == nil {
+			res.SetError(errNotOnline, cmds.ErrClient)
+			return
+		}
+
+		infos := make(map[string]interface{})
+		cmgr := n.PeerHost.ConnManager()
+		switch cmgr := cmgr.(type) {
+		case nil:
+			infos["type"] = "<nil>"
```

Comment:
Is distinguishing between these two cases really useful? Users aren't going to know what `<nil>` means.

Reply:
They shouldnt ever see "nil", Everything currently will either return a NullConnMgr or a BasicConnMgr. So maybe I should remove it?

Reply:
Probably. I'd fail with an error if this isn't supposed to happen.